python/pyproject.toml

```diff
@@ -15,15 +15,14 @@
 [build-system]
 build-backend = "setuptools.build_meta"
 requires = [
-    "wheel",
-    "setuptools>=61.0.0",
-    "cython>=0.29,<0.30",
-    "scikit-build>=0.13.1",
     "cmake>=3.23.1,!=3.25.0",
+    "cython>=0.29,<0.30",
     "ninja",
-    "cuda-python>=11.7.1,<12.0",
-    "tomli; python_version < '3.11'",
-]
+    "scikit-build>=0.13.1",
+    "setuptools",
+    "tomli",
```

Comment:
Looks like the Python version constraint is dropped. We don't support Python 3.11 now. Though would we want this constraint?

Reply:
The question is whether there is a Python version constraint syntax that is respected by both conda and pip; conda doesn't support [PEP 508-style environment markers](https://peps.python.org/pep-0508/#environment-markers), does it? If not, we would need to specify tomli twice in dependencies, once for pyproject.toml and once for the conda environment.yaml files.

Reply:
Agreed, I think given that it won't conflict with tomllib the easiest solution is to just include it unconditionally and then remove it when we drop support for Python < 3.11

Reply:
This solution is fine -- but if the problem was deeper, I would be fine with the alternative of specifying it twice as well.
> If not, we would need to specify tomli twice in dependencies, once for pyproject.toml and once for the conda environment.yaml files.